pkg/backend/httpstate/backend.go

```diff
@@ -526,6 +526,18 @@ func (b *cloudBackend) CreateStack(ctx context.Context, stackRef backend.StackRe
 	stack := newStack(apistack, b)
 	fmt.Printf("Created stack '%s'.\n", stack.Ref())
 
+	// Some organizations have soft limits associated with them. If we created the stack but it caused the limit to
+	// be exceeded, print a helpful warning. Note that the stack WAS created, so the overall operation should
+	// not fail.
+	if createResp.MaxStackLimit != nil && createResp.StackCount > *createResp.MaxStackLimit {
```

Comment:
I wish there was a way to simply do this in the service. Having the client know about service-based limits feels odd to me. Furthermore, in the future, if they turn into hard limits, this approach will no longer work and we'll need a different way to convey the information (presumably a 4xx with a meaningful body).

An alternative would be to return a 200 from the service, but to include "advisory" information in the body (e.g., a warning field or something). Then the client could simply serve up the warning text without needing to understand the concept of stack limits (or whatever limits we happen to introduce down the road).

Curious what you and @chrsmith think.

Reply:
Chris and I did talk about this briefly.  We do have a (presently unused) RequestError type that we envision using to provide more context for hard errors in the future.  Originally, Chris thought we should just add a field to the response of this type, and I talked him out of it, because this didn’t feel like a warning-ish thing to me. 

If we just want to include an array of strings which are messages in the response (instead of the proposed approach of adding both the current stack count and the maximum), I’m fine with that.

I do wonder if we want to continue to at least include this information so the CLI could display information whenever you create a new stack showing how many of your soft limit you are using.  But that once again requires the CLI to have some sort of innate understanding of states the system could be in, so it can decide what to print.

I have a slight preference with pushing this logic from the client (obviously, since I did it this way) because these feel like presentation concerns and so I want to defer the logic to the presentation layer.

But, if you and Chris are more comfortable with just “here’s a bunch of warnings, go print them” as a general pattern for our APIs, I’m happy to get on board.

Reply:
Matt and I spoke about this for a while, and even started with an optional "warnings" field in the REST response. This seemed like the better approach, at least in the short-term. And given the importance of the message returned, giving the CLI more control in its presentation did seem like a really good idea.

So I'm inclined to keep this as-is. But if we do want a more flexible approach, we should just check for an optional HTTP header in the response, e.g. `PULUMI-WARNING`, and if present, the CLI should just print it to STDERR. Without any context for what the warning actually means. That would allow us to add the data to any response without needing to change the shape of API types.